File: engine/model/src/main/scala/tools/packet/alliance/DisbandAlliance.scala
package tools.packet.alliance

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class DisbandAlliance(private var _allianceId: Int) extends PacketInput {
  def allianceId: Int = _allianceId

  override def opcode(): SendOpcode = SendOpcode.ALLIANCE_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/alliance/DisbandAlliance.java
package tools.packet.alliance;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record DisbandAlliance(Integer allianceId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.ALLIANCE_OPERATION;
   }
}
